src/bytes.rs

```diff
@@ -705,8 +360,13 @@ impl Bytes {
     /// ```
     ///
     /// [`split_off`]: #method.split_off
+    #[inline]
     pub fn truncate(&mut self, len: usize) {
-        self.inner.truncate(len);
+        if len >= self.len {
+            self.len = 0;
```

Comment:
I don't think this is correct. The documentation says `truncate` should have no effect in this case.